File: strong.py
import math

def is_strong_number(num):
    return sum(math.factorial(int(digit)) for digit in str(num)) == num

def print_strong_numbers_in_range(start, end):
    print(f"Strong numbers between {start} and {end}:")
    strong_numbers = [i for i in range(start, end + 1) if is_strong_number(i)]
    if strong_numbers:
        for num in strong_numbers:
            print(num)
    else:
        print("No strong numbers found in this range.")

print_strong_numbers_in_range(1, 5000)

File: strong.java
public class strong{

    // Function to calculate factorial of a number
    public static int factorial(int n) {
        if (n == 0 || n == 1) {
            return 1;
        }
        int fact = 1;
        for (int i = 2; i <= n; i++) {
            fact *= i;
        }
        return fact;
    }

    // Function to check if a number is strong
    public static boolean isStrongNumber(int number) {
        int originalNumber = number;
        int sumOfFactorials = 0;
        int temp = number;

        while (temp > 0) {
            int digit = temp % 10;
            sumOfFactorials += factorial(digit);
            temp /= 10;
        }
        return sumOfFactorials == originalNumber;
    }

    public static void main(String[] args) {
        System.out.println("Strong numbers from 1 to 5000 are:");
        for (int i = 1; i <= 5000; i++) {
            if (isStrongNumber(i)) {
                System.out.println(i);
            }
        }
    }
}